src/github.com/getlantern/flashlight/config/cloud.go

```diff
@@ -0,0 +1,152 @@
+package config
+
+import (
+	"compress/gzip"
+	"fmt"
+	"io/ioutil"
+	"math/rand"
+	"net/http"
+	"time"
+
+	"github.com/getlantern/fronted"
+	"github.com/getlantern/proxiedsites"
+	"github.com/getlantern/yaml"
+
+	"github.com/getlantern/flashlight/client"
+)
+
+// cloudConfig is the in memory representation of cloud.yaml
+type cloudConfig struct {
+	// To simplify, just use an ClientConfig object here.
+	// Only those fields existed in cloud.yaml will take effect.
+	Client       *client.ClientConfig
+	ProxiedSites *proxiedsites.Config
+	TrustedCAs   []*CA
+}
+
+const (
+	CloudConfigPollInterval = 1 * time.Minute
+	cloudflare              = "cloudflare"
+	etag                    = "X-Lantern-Etag"
+	ifNoneMatch             = "X-Lantern-If-None-Match"
+)
+
+var (
+	cloudConfigChanged chan bool = make(chan bool)
+)
+
+func emptyCloudConfig() *cloudConfig {
+	return &cloudConfig{
+		Client: &client.ClientConfig{
+			FrontedServers: []*client.FrontedServerInfo{},
+			ChainedServers: map[string]*client.ChainedServerInfo{},
+			MasqueradeSets: map[string][]*fronted.Masquerade{},
+		},
+		ProxiedSites: &proxiedsites.Config{
+			Delta: &proxiedsites.Delta{},
+		},
+		TrustedCAs: []*CA{},
+	}
+}
+
+func startCloudPoll() {
+	go func() {
+		for {
+			time.Sleep(cloudPollSleepTime())
+			cloudPoll()
+		}
+	}()
+}
+
+func cloudPoll() {
+	cfg := localCfg.Load().(*Config)
+	b, err := fetchCloudConfig(cfg.CloudConfig)
+	if err != nil {
+		log.Errorf("Error fetch cloud config: %s", err)
+		return
+	}
+	if b == nil {
+		return
+	}
+	newCfg := emptyCloudConfig()
+	if err = newCfg.fromBytes(b); err != nil {
+		log.Errorf("Error parse cloud config: %s", err)
+		return
+	}
+	newCfg.Client.SortFrontedServers()
```

Comment:
We have only 1 fronted server in default config, so I move it here, in case there are more than 1 in cloud.yaml